docs/sources/tempo/metrics-generator/cardinality.md

```diff
@@ -0,0 +1,83 @@
+---
+aliases:
+- /docs/tempo/latest/metrics-generator/cardinality
+title: Cardinality
+menuTitle: Cardinality
+description: What is cardinality and how it is impacted by metrics generation?
+weight: 100
+---
+
+# Cardinality
+
+Cardinality refers to the total combination of key/value pairs, such as labels and label values for a given metric series or log stream, and how many unique combinations they generate.
+For more information on cardinality, see the [What are cardinality spikes and why do they matter?](/blog/2022/02/15/what-are-cardinality-spikes-and-why-do-they-matter/) blog post.
+
+Because writes are series, high cardinality does not make a big difference. However, cardinality can have a major impact on querying where you iterate over a lot more things.
+
+Cardinality can pose a problem when you have many services.
+There isn't a direct formula or solution to this issue.
+The following guide should help estimate the cardinality that metrics generator will create.
+
+## Traces collection and metrics
+
+Tempo’s server-side metrics generation adds functionality to the collection of traces by creating Prometheus-based metrics that track a variety of metrics such as:
+
+- Total span call counts
+- Span latency histograms
+- Total span size count
+
+The metrics generator also allows you to create metrics which define the relationship between services via edges and nodes.
+Each of these metrics are queryable via a set of Prometheus labels (key/value pairs).
+
+Each new value for a label increases the number of active series associated with a metric. (To learn more about active series, see the [Trace active series]({{< relref "./active-series" >}}) documentation.)
+
+This is also known as an increase in cardinality, and the number of active series generated for a metric is directly proportional to the number of labels that exist for that metrics alongside the number of values each label has added.
+
+In a non-modified instance of the metrics generator, there are a small number of labels that are added automatically.
+Because labels like `span_kind` and `status_code` only have a small number of valid values, the largest variable for the number of active series produced for each metric will depend on the number of service names and span names associated with trace spans.
+
+The metrics generator can also be configured to also add extra labels on metrics, using span attribute key/value pairs which are mapped directly to these labels see the [custom span attribute documentation]({{< relref "../configuration#metrics-generator" >}}).
+
+Be careful when configuring custom attributes: the greater the number of values seen in a specific attribute, the greater the number of active series will be produced.
+
+For example, if adding a custom attribute that includes unique customer IDs as a metrics label, if you have 100 customers this could multiple the number of active series generated by 100 (for example, going from 25,000 active series to 2.5M).
```

Comment:
```suggestion
For example, if adding a custom attribute that includes unique customer IDs as a metrics label, if you have 100 customers this could potentially multiple the number of active series generated by up to 100 (for example, going from 25,000 active series to 2.5M).
```